cpp/src/arrow/filesystem/s3fs.h

```diff
@@ -62,10 +62,12 @@ struct ARROW_EXPORT S3Options {
   void ConfigureAnonymousCredentials();
 
   /// Configure with explicit access and secret key.
-  void ConfigureAccessKey(const std::string& access_key, const std::string& secret_key);
+  void ConfigureAccessKey(const std::string& access_key, const std::string& secret_key,
+                          const std::string& session_token = "");
 
   std::string GetAccessKey() const;
   std::string GetSecretKey() const;
+  std::string GetSessionToken() const;
```

Comment:
Please cover the new API with tests on the C++ side.

Reply:
I added C++ tests to `s3fs_test.cc` to cover these changes.  Unfortunately, adding support for STS-like temporary credentials to Minio involves addition of another dependency, Keycloak.  I spent a bit of time reading relevant docs but could not seem to find a programmatic way to replicate the desired behavior (the Keycloak guides I found were very UI-driven).

Reply:
I think it's ok not to have any deeper test.